promql/parser/ast.go

```diff
@@ -49,9 +49,9 @@ type Node interface {
 type Statement interface {
 	Node
 
-	// stmt ensures that no other type accidentally implements the interface
+	// PromQLStmt ensures that no other type accidentally implements the interface
```

Comment:
this comment seems inaccurate/unnecessary now

Reply:
The comment is needed to explain why this dead code is here.

Reply:
@brian-brazil my point was that the wording, specifically the use of `accidentally`, is confusing and inaccurate in my opinion now that this function is exported.

Reply:
I'm not seeing that, the accident would be having a type that didn't mean to implement Statement inadvertently implementing Statement.